app/src/main/java/Fragments/ProfileFragment.java

```diff
@@ -373,4 +258,123 @@ public void onFailure(@NonNull Exception exception) {
+
+    // SET Auth State Listener
+        authListener = new FirebaseAuth.AuthStateListener() {
+            @Override
+            public void onAuthStateChanged(@NonNull FirebaseAuth firebaseAuth) {
+                if(authRef.getCurrentUser() != null) {
+                    // Display menu save option
+                    setHasOptionsMenu(true);
+
+                    name = authRef.getCurrentUser().getDisplayName();
+                    email = authRef.getCurrentUser().getEmail();
+                    //photoUrl = user.getPhotoUrl();
+
+                /*if (name != null) {
+                    name_et.setText(name);
+                }*/
+
+                    // TODO: Check if user's email is verified?
+                    email_et.setText(email);
+
+               /* if (photoUrl != null) {
+                    new DownloadImageTask(photo_iv).execute(photoUrl.toString());
+                }*/
+
+                    name_et.setVisibility(View.VISIBLE);
+                    email_et.setVisibility(View.VISIBLE);
+                    phoneNumber_et.setVisibility(View.VISIBLE);
+                    photo_iv.setVisibility(View.VISIBLE);
+
+
+                    loggedOut_tv.setVisibility(View.GONE);
+                    login_b.setVisibility(View.GONE);
+                    signup_b.setVisibility(View.GONE);
+
+                    ZIP_et.setVisibility(View.VISIBLE);
+                    updatePhoto_ib.setVisibility(View.GONE);
+                    update_ib.setVisibility(View.VISIBLE);
+                    updatePhoto_ib.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View view) {
+                            updatePhoto();
+                        }
+                    });
+
+                // SET Data Listener
+                    myUID = myUser.child(authRef.getCurrentUser().getUid());
+                    myUID.addValueEventListener(new ValueEventListener() {
+                        @Override
+                        public void onDataChange(DataSnapshot dataSnapshot) {
+                            String imgUrl = dataSnapshot.child("ImageURL").getValue(String.class);
+                            //Toast.makeText(getActivity(), "OnStart " + imgUrl, Toast.LENGTH_LONG).show();
+                            if(imgUrl != null)
+                                new DownloadImageTask(photo_iv).execute(imgUrl);
+
+                            String name = dataSnapshot.child("FirstName").getValue(String.class) + " " + dataSnapshot.child("LastName").getValue(String.class);
+                            name_et.setText(name);
+
+                            String phoneNbr = dataSnapshot.child("PhoneNumber").getValue(String.class);
+                            phoneNumber_et.setText(phoneNbr);
+
+                            String ZIP = dataSnapshot.child("ZIPCode").getValue(String.class);
+                            ZIP_et.setText(ZIP);
+                        }
```

Comment:
Really great new implementation to update data. I did not see that possibility when coding it. Good job.